src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size < array.length) {
+            array[size] = value;
+            size++;
+        } else {
+            array = grow(array);
+            array[size] = value;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < size && index >= 0) {
+            if (size == array.length) {
+                array = grow(array);
+            }
+            System.arraycopy(array, index, array, index + 1, array.length - index - 1);
+            array[index] = value;
+            size++;
+        } else if (index == size) {
+            this.add(value);
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + size);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (list.size() + size > array.length) {
+            array = grow(array);
+        }
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
```

Comment:
```suggestion
        checkIfIndexFits(index);
```

Create separate method